src/dataframe/dataframe.jl

```diff
@@ -227,19 +236,23 @@ function DataFrame(pairs::AbstractVector{<:Pair}; makeunique::Bool=false,
     if isempty(pairs)
         return DataFrame()
     else
-        return DataFrame(pairs..., makeunique=makeunique, copycols=copycols)
+        if !(all(((k,v),) -> k isa Symbol, pairs) || all(((k,v),) -> k isa AbstractString, pairs))
+            throw(ArgumentError("All column names must be Symbols or strings"))
+        end
+        colnames = [Symbol(k) for (k,v) in pairs]
+        columns = Any[v for (k,v) in pairs]
```

Comment:
Doesn't this avoid a copy?
```suggestion
        columns = AbstractVector[v for (k,v) in pairs]
```

Reply:
This and below is incorrect as `v` does not have to be a vector - we allow pseudo broadcasting here. Here is an example on 0.21:
```
julia> using DataFrames

julia> DataFrame([:A => 1, :B => 11:12])
2×2 DataFrame
│ Row │ A     │ B     │
│     │ Int64 │ Int64 │
├─────┼───────┼───────┤
│ 1   │ 1     │ 11    │
│ 2   │ 1     │ 12    │
```
I think we should keep it so that `DataFrame(vec_of_pairs)` and `DataFraem(vec_of_pairs...)` behave the same way.